### PR TITLE
Calibration example

### PR DESCRIPTION
**Pull request by wiseodd**
I put the example in a dedicated directory `./examples` since otherwise it will be too untidy at the root directory. Please let me know your thoughts.

### CODE REVIEW

**Comment by wiseodd** [comment]
This is for addressing #49

File: examples/calibration_example.md [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+``` python
+model = wrn.WideResNet(16, 4, num_classes=10).cuda().eval()
+
+# Download pre-trained model if necessary
+if not os.path.isfile('CIFAR10_plain.pt'):
+    if not os.path.exists('./temp'):
+        os.makedirs('./temp')
+
+    urllib.request.urlretrieve('https://nc.mlcloud.uni-tuebingen.de/index.php/s/2PBDYDsiotN76mq/download', './temp/CIFAR10_plain.pt')
```
Comment (AlexImmer) [inline examples/calibration_example.md:43]: Maybe these 4 lines can be hidden away in the `helper` methods and wrapped in `download_model_temp()` or something alike?

File: examples/calibration_example.md [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+We are now ready to see how calibrated is the model. The metric we use is the expected calibration error (ECE, Naeni et al., AAAI 2015). Note that lower values are better for this metric.
+
+First, let us inspect the MAP model. We shall use the [`netcal`](https://github.com/fabiankueppers/calibration-framework) library to easily compute the ECE.
+
+``` python
+probs_map = predict(test_loader, model, laplace=False)
+acc_map = (probs_map.argmax(-1) == targets).mean()
+ece_map = ECE(bins=15).measure(probs_map, targets)
+
+print(f'[MAP] Acc.(🠕): {acc_map:.1%}; ECE(🠗): {ece_map:.1%}')
```
Comment (AlexImmer) [inline examples/calibration_example.md:75]: What is the meaning of "🠕" in the brackets? Is it some unicode thing that doesn't render properly?

File: examples/helper/dataloaders.py [hunk 3]
```diff
@@ -0,0 +1,131 @@
+import torch
```
Comment (AlexImmer) [inline examples/helper/dataloaders.py:1]: Why are all these data sets available if we only need one?

File: examples/helper/wideresnet.py [hunk 4]
```diff
@@ -0,0 +1,111 @@
+import math
```
Comment (AlexImmer) [inline examples/helper/wideresnet.py:1]: Is this model taken from somewhere or just standard? Just so people know about potential differences to a standard WRN.

**Review by AlexImmer — COMMENTED** [review]
I recompiled the documentation to integrate the new example already. You can probably also view it locally. For detailed questions/feedback, see my comments.

**Comment by wiseodd** [comment]
@AlexImmer updated. Please check.
